project/lebr0nli/README.rst

```diff
@@ -0,0 +1,184 @@
+However, the Python version, `py-rbush`, is an outdated and dead project, with its last commit made over six years ago.
+This project seeks to rewrite `py-rbush` from the ground up, aiming to bring the performance and capabilities of the original `rbush` to Python, while taking advantage of Python-specific optimizations.
+Also I will add some missing features of the current `py-rbush` from the `rbush`.
+
+Problem to Solve
+====================
+
+Although `numpy` is used in `py-rbush`, I believe there are areas where performance can still be improved.
+The primary focus of this project will be:
+
+1. Rewriting the core algorithms to enhance performance while maintaining the simplicity of use.
+
+2. `py-rbush` didn't implement all the features of the original `rbush`, so I will add those missing features.
+
+Prospective Users
+=================
+The updated version of `py-rbush` will be valuable for developers and researchers in fields that require fast, scalable spatial indexing, such as:
+
+* Geographic Information Systems (GIS)
+* Computer Vision and Image Processing
+* Game Development
+* Computational Geometry
+
+System Architecture
+===================
+
+`py-rbush` will consist of two main components:
+
+1. **C++ Backend:** The core R-Tree implementation will be written in C++ for performance reasons. This will be wrapped using `pybind11` to provide a Python interface.
+2. **Python Frontend:** The Python frontend will provide a user-friendly API for interacting with the R-Tree. This will include methods for inserting, deleting, and querying spatial data.
+
+API Description
+===============
+
+The library will follow the design and functionality of the original `prbush`. The core API will include:
+
+1. **Creating a Tree**
+
+   .. code-block:: python
+
+      tree = RBush(max_entries=9)
+
+   An optional `max_entries` parameter defines the maximum number of entries in each tree node. A value of `9` is used by default, but this can be adjusted based on performance needs.
+
+2. **Adding Data**
+
+   Insert an item:
+
+   .. code-block:: python
+
+      item = {
+          'minX': 20,
+          'minY': 40,
+          'maxX': 30,
+          'maxY': 50,
+          'data': 'example'
+      }
+      tree.insert(item)
+
+3. **Removing Data**
+
+   Remove a previously inserted item:
+
+   .. code-block:: python
+
+      tree.remove(item)
+
+   If a custom comparison is needed (e.g., comparing by specific fields), a function can be provided:
+
+   .. code-block:: python
+
+      tree.remove(item_copy, lambda a, b: a['id'] == b['id'])
+
+   Remove all items:
+
+   .. code-block:: python
+
+      tree.clear()
+
+4. **Data Format**
+
+   By default, items are expected to have `minX`, `minY`, `maxX`, and `maxY` properties. This format can be customized by overriding methods:
```

Comment:
Do you want to limit to two-dimensional space?

Reply:
Yes, I plan to start with two-dimensions since I don't have much experience in this area, and I'm unsure of the difficulty involved.
However, if the implementation goes smoothly and doesn't require too much effort, I'll try to extend it to three dimensions.